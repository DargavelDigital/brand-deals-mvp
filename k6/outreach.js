import { sleep } from 'k6'
import { postJson } from './common.js'

export const options = {
  thresholds: {
    http_req_duration: ['p(95)<1500'],
    checks: ['rate>0.99']
  },
  scenarios: { 
    burst: { 
      executor: 'per-vu-iterations', 
      vus: 10, 
      iterations: 5 
    } 
  }
}

export default function() {
  postJson('/api/outreach/start', {
    workspaceId: 'demo-workspace',
    pauseFirstSend: true,
    sequence: { name: 'CI load', steps: [] }
  })
  sleep(0.1)
}
