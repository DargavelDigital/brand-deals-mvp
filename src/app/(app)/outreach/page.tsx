'use client'

import { useState } from 'react'
import { Col } from '@/ui/containers'
import Button from '@/components/ui/Button'

interface EmailTemplate {
  id: string
  name: string
  description: string
  variables: string[]
  lastUsed?: Date
}

interface Outreach {
  id: string
  brand: string
  template: string
  status: 'sent' | 'delivered' | 'opened' | 'clicked' | 'replied'
  sentAt: Date
  response?: string
}

const mockTemplates: EmailTemplate[] = [
  {
    id: '1',
    name: 'Intro Email',
    description: 'Initial outreach to introduce yourself and your content',
    variables: ['{{creatorName}}', '{{brandName}}', '{{contentType}}'],
    lastUsed: new Date('2024-01-15')
  },
  {
    id: '2',
    name: 'Follow-up Email',
    description: 'Gentle reminder after initial contact',
    variables: ['{{creatorName}}', '{{brandName}}', '{{previousEmail}}'],
    lastUsed: new Date('2024-01-10')
  }
]

const mockOutreach: Outreach[] = [
  {
    id: '1',
    brand: 'Nike',
    template: 'Intro Email',
    status: 'replied',
    sentAt: new Date('2024-01-15'),
    response: 'Thanks for reaching out! We\'d love to discuss a potential partnership.'
  },
  {
    id: '2',
    brand: 'Apple',
    template: 'Intro Email',
    status: 'opened',
    sentAt: new Date('2024-01-14')
  }
]

const getStatusColor = (status: string) => {
  switch (status) {
    case 'sent': return ''
    case 'delivered': return ''
    case 'opened': return ''
    case 'clicked': return ''
    case 'replied': return ''
    default: return ''
  }
}

const formatDate = (date: Date) => {
  return date.toLocaleDateString()
}

export default function OutreachPage() {
  const [templates] = useState<EmailTemplate[]>(mockTemplates)
  const [outreach] = useState<Outreach[]>(mockOutreach)

  return (
    <div>
      <div>
        <h1>Outreach Management</h1>
        <p>Manage your email campaigns and templates</p>
      </div>

      <div>
        <Col>
          <div>
            <h2>Email Templates</h2>
            <div>
              {templates.map((template) => (
                <div key={template.id}>
                  <h3>{template.name}</h3>
                  <p>{template.description}</p>
                  <p>
                    Variables: {template.variables.join(', ')}
                  </p>
                  <div>
                    {template.variables.map((variable) => (
                      <span key={variable}>
                        {variable}
                      </span>
                    ))}
                  </div>
                  {template.lastUsed && (
                    <p>
                      Last used: {formatDate(template.lastUsed)}
                    </p>
                  )}
                  <div>
                    <Button>
                      Edit Template
                    </Button>
                    <Button>
                      Use Template
                    </Button>
                  </div>
                </div>
              ))}
            </div>
          </div>
        </Col>

        <Col>
          <div>
            <h2>Recent Outreach</h2>
            <div>
              {outreach.map((outreachItem) => (
                <div key={outreachItem.id}>
                  <div>
                    <span>{outreachItem.brand}</span>
                    <span>
                      {outreachItem.status}
                    </span>
                  </div>
                  <p>{outreachItem.template}</p>
                  <p>{formatDate(outreachItem.sentAt)}</p>
                  {outreachItem.response && (
                    <p>"{outreachItem.response}"</p>
                  )}
                </div>
              ))}
            </div>
          </div>
        </Col>
      </div>
    </div>
  )
}
