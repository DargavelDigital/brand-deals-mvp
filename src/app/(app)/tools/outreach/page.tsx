'use client'

import ToolPage from '@/components/tools/ToolPage'
import { startOutreach } from '@/lib/api'

export default function OutreachTool() {
  return (
    <ToolPage
      title="Outreach"
      subtitle="Start a 3â€“4 touch sequence with reply detection."
      primaryLabel="Start Outreach"
      onPrimary={() => startOutreach()}
      showAdvance
    />
  )
}
