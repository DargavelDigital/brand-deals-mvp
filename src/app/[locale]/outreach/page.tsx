'use client'

import { Section } from "@/components/ui/Section";
import { Card } from "@/components/ui/Card";
import { Input } from "@/components/ui/Input";
import { Button } from "@/components/ui/Button";

export default function OutreachPage() {
  return (
    <Section title="Outreach" description="Sequences and templates">
      <div className="space-y-6">
        {/* Editor card */}
        <Card className="p-6 space-y-4">
          <div>
            <label className="block text-sm font-medium mb-2">Subject Line</label>
            <Input placeholder="Enter email subject..." />
          </div>
          
          <div>
            <label className="block text-sm font-medium mb-2">Email Template</label>
            <textarea 
              className="min-h-[120px] w-full rounded-md border border-[var(--border)] px-3 py-2 focus-visible:outline-2 focus-visible:outline-[var(--accent)]"
              placeholder="Write your email template here..."
            />
          </div>
          
          <div>
            <label className="block text-sm font-medium mb-2">Sequence Name</label>
            <Input placeholder="e.g., Initial Outreach, Follow-up 1" />
          </div>
        </Card>

        {/* Actions */}
        <div className="flex items-center gap-3">
          <Button>Save Template</Button>
          <Button variant="secondary">Preview</Button>
          <Button variant="ghost">Load Template</Button>
        </div>

        {/* Templates list */}
        <Card className="p-6">
          <h3 className="text-lg font-semibold mb-4">Saved Templates</h3>
          <div className="space-y-3">
            <div className="flex items-center justify-between p-3 border border-[var(--border)] rounded-md">
              <div>
                <div className="font-medium">Initial Outreach</div>
                <div className="text-sm text-[var(--muted)]">First contact with potential brand partners</div>
              </div>
              <div className="flex items-center gap-2">
                <Button variant="ghost" size="sm">Edit</Button>
                <Button variant="ghost" size="sm">Use</Button>
              </div>
            </div>
            
            <div className="flex items-center justify-between p-3 border border-[var(--border)] rounded-md">
              <div>
                <div className="font-medium">Follow-up</div>
                <div className="text-sm text-[var(--muted)]">Gentle reminder after initial contact</div>
              </div>
              <div className="flex items-center gap-2">
                <Button variant="ghost" size="sm">Edit</Button>
                <Button variant="ghost" size="sm">Use</Button>
              </div>
            </div>
          </div>
        </Card>
      </div>
    </Section>
  );
}
