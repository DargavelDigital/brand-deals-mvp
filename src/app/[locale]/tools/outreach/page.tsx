'use client'
import OutreachPage from '@/components/outreach/OutreachPage'
import { isToolEnabled } from '@/lib/launch'
import { ComingSoon } from '@/components/ComingSoon'
import PageShell from '@/components/PageShell'

export default function Page(){ 
  const enabled = isToolEnabled("outreach")
  
  if (!enabled) {
    return (
      <PageShell title="Outreach" subtitle="Manage your outreach campaigns and track responses.">
        <div className="mx-auto max-w-md">
          <ComingSoon
            title="Outreach"
            subtitle="This tool will be enabled soon. The page is visible so you can navigate and preview the UI."
          />
        </div>
      </PageShell>
    )
  }
  
  return <OutreachPage/> 
}
