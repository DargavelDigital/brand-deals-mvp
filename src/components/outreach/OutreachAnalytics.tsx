'use client'

import { useState, useEffect } from 'react'
import { Card } from '@/components/ui/Card'
import { Badge } from '@/components/ui/Badge'
import { Select } from '@/components/ui/Select'
import { BarChartComponent } from '@/ui/charts'
import { Button } from '@/components/ui/Button'

interface AnalyticsData {
  funnel: {
    sent: number
    opened: number
    clicked: number
    replied: number
    bounced: number
  }
  rates: {
    openRate: number
    clickRate: number
    replyRate: number
  }
  medianTimeToReply: number
  chartData: Array<{
    name: string
    value: number
    color: string
  }>
  topSubjects: Array<{
    subject: string
    totalSent: number
    totalOpened: number
    totalReplied: number
    openRate: number
    replyRate: number
  }>
  period: string
}

export default function OutreachAnalytics() {
  const [analytics, setAnalytics] = useState<AnalyticsData | null>(null)
  const [loading, setLoading] = useState(true)
  const [error, setError] = useState<string | null>(null)
  const [period, setPeriod] = useState('30')

  useEffect(() => {
    fetchAnalytics()
  }, [period])

  const fetchAnalytics = async () => {
    try {
      setLoading(true)
      const response = await fetch(`/api/outreach/analytics?days=${period}`)
      
      if (response.ok) {
        const result = await response.json()
        setAnalytics(result.data)
        setError(null)
      } else {
        setError('Failed to fetch analytics')
      }
    } catch (err) {
      setError('Error fetching analytics')
      console.error('Error:', err)
    } finally {
      setLoading(false)
    }
  }

  if (loading) {
    return (
      <div className="space-y-6">
        <div className="animate-pulse">
          <div className="h-8 bg-gray-200 rounded w-1/4 mb-4"></div>
          <div className="grid grid-cols-4 gap-4 mb-6">
            {[1, 2, 3, 4].map((i) => (
              <div key={i} className="h-20 bg-gray-200 rounded"></div>
            ))}
          </div>
          <div className="h-64 bg-gray-200 rounded"></div>
        </div>
      </div>
    )
  }

  if (error) {
    return (
      <div className="space-y-6">
        <div className="card p-4 border-[var(--error)] bg-[var(--tint-error)] text-[var(--error)]">
          {error}
          <Button 
            variant="secondary" 
            size="sm" 
            onClick={fetchAnalytics}
            className="ml-3"
          >
            Retry
          </Button>
        </div>
      </div>
    )
  }

  if (!analytics) {
    return (
      <div className="space-y-6">
        <div className="text-center text-[var(--muted-fg)] py-12">
          No analytics data available
        </div>
      </div>
    )
  }

  return (
    <div className="space-y-6">
      {/* Header with period selector */}
      <div className="flex items-center justify-between">
        <div>
          <h2 className="text-xl font-semibold">Sequence Analytics</h2>
          <p className="text-[var(--muted-fg)]">Performance metrics for your outreach sequences</p>
        </div>
        <div className="flex items-center gap-2">
          <label className="text-sm text-[var(--muted-fg)]">Period:</label>
          <Select
            value={period}
            onChange={(e) => setPeriod(e.target.value)}
            className="w-32"
          >
            <option value="7">7 days</option>
            <option value="30">30 days</option>
            <option value="90">90 days</option>
          </Select>
        </div>
      </div>

      {/* KPI Row */}
      <div className="grid grid-cols-1 md:grid-cols-4 gap-4">
        <Card className="p-4 text-center">
          <div className="text-2xl font-bold text-[var(--accent)]">
            {analytics.funnel.sent.toLocaleString()}
          </div>
          <div className="text-sm text-[var(--muted-fg)]">Emails Sent</div>
        </Card>
        
        <Card className="p-4 text-center">
          <div className="text-2xl font-bold text-[var(--success)]">
            {analytics.rates.openRate.toFixed(1)}%
          </div>
          <div className="text-sm text-[var(--muted-fg)]">Open Rate</div>
        </Card>
        
        <Card className="p-4 text-center">
          <div className="text-2xl font-bold text-[var(--warn)]">
            {analytics.rates.replyRate.toFixed(1)}%
          </div>
          <div className="text-sm text-[var(--muted-fg)]">Reply Rate</div>
        </Card>
        
        <Card className="p-4 text-center">
          <div className="text-2xl font-bold text-[var(--brand-600)]">
            {analytics.medianTimeToReply > 0 
              ? `${analytics.medianTimeToReply.toFixed(1)}h`
              : 'N/A'
            }
          </div>
          <div className="text-sm text-[var(--muted-fg)]">Median Reply Time</div>
        </Card>
      </div>

      {/* Funnel Chart */}
      <Card className="p-6">
        <h3 className="text-lg font-medium mb-4">Email Funnel</h3>
        <div className="h-64">
          <BarChartComponent data={analytics.chartData} height={256} />
        </div>
        <div className="mt-4 text-sm text-[var(--muted-fg)] text-center">
          Showing data for the last {analytics.period}
        </div>
      </Card>

      {/* Top Performing Subjects */}
      <Card className="p-6">
        <h3 className="text-lg font-medium mb-4">Top Performing Subjects</h3>
        <div className="overflow-x-auto">
          <table className="w-full text-sm">
            <thead>
              <tr className="border-b border-[var(--border)]">
                <th className="text-left py-2 font-medium">Subject</th>
                <th className="text-right py-2 font-medium">Sent</th>
                <th className="text-right py-2 font-medium">Opened</th>
                <th className="text-right py-2 font-medium">Replied</th>
                <th className="text-right py-2 font-medium">Open Rate</th>
                <th className="text-right py-2 font-medium">Reply Rate</th>
              </tr>
            </thead>
            <tbody>
              {analytics.topSubjects.map((subject, index) => (
                <tr key={index} className="border-b border-[var(--border)] hover:bg-[var(--muted)]/20">
                  <td className="py-2 font-medium truncate max-w-xs" title={subject.subject}>
                    {subject.subject}
                  </td>
                  <td className="py-2 text-right">{subject.totalSent}</td>
                  <td className="py-2 text-right">{subject.totalOpened}</td>
                  <td className="py-2 text-right">{subject.totalReplied}</td>
                  <td className="py-2 text-right">
                    <Badge variant="outline" className="text-xs">
                      {subject.openRate.toFixed(1)}%
                    </Badge>
                  </td>
                  <td className="py-2 text-right">
                    <Badge variant="outline" className="text-xs">
                      {subject.replyRate.toFixed(1)}%
                    </Badge>
                  </td>
                </tr>
              ))}
            </tbody>
          </table>
        </div>
        {analytics.topSubjects.length === 0 && (
          <div className="text-center text-[var(--muted-fg)] py-8">
            No subject data available for this period
          </div>
        )}
      </Card>
    </div>
  )
}
