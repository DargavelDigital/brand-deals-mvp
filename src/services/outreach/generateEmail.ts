import { aiInvoke } from '../../ai/invoke';
import { isFlagEnabled } from '../../lib/flags';
import { toneToOptions } from './toneAdapter';
import type { StyleTone, StyleBrevity } from '../../ai/types';
import type { OutreachTone } from '../../types/outreach';

export interface EmailGenerationInput {
  creator: {
    name: string;
    niche: string;
    country: string;
    followers: number;
  };
  brand: {
    name: string;
    industry: string;
    category: string;
    country: string;
  };
  mediaPackUrl?: string;
  audit?: {
    audience: {
      followers: number;
      topCountries: string[];
    };
    content: {
      formats: string[];
      avgEngagement: number;
    };
  };
}

export interface EmailGenerationOutput {
  subject: string;
  body: string;
  toneUsed: string;
  reasoning: string;
  placeholders?: {
    brand_name: string;
    creator_name: string;
    media_pack_url: string;
  };
}

export async function generateOutreachEmail(
  workspaceFlags: any,
  input: EmailGenerationInput & { sequence?: { settings?: { tone?: OutreachTone } } },
  opts?: { 
    tone?: StyleTone; 
    brevity?: StyleBrevity;
  }
): Promise<EmailGenerationOutput> {
  const enabled = isFlagEnabled?.('OUTREACH_TONES', workspaceFlags) ?? true; // default on if flag infra missing
  const tone = input.sequence?.settings?.tone ?? 'professional';

  if (!enabled) {
    // Legacy fallback if you still need it
    return {
      subject: `Introduction from ${input.brand.name}`,
      body: `Hi ${input.creator.name},

I hope this email finds you well! I'm reaching out from ${input.brand.name} because I believe there's a great opportunity for collaboration.

Your content in the ${input.creator.niche} space and your engaged audience of ${input.creator.followers.toLocaleString()} followers makes you an ideal partner for ${input.brand.industry} brand.

${input.mediaPackUrl ? `I've attached our media pack here: ${input.mediaPackUrl}` : "I'd love to share more about our partnership opportunities."}

Would you be interested in a quick call to discuss how we could work together?

Best regards,
[Your Name]
${input.brand.name} Partnership Team`,
      toneUsed: 'basic',
      reasoning: 'Basic email template used. Enable OUTREACH_TONES for AI-powered personalization.'
    };
  }

  const toneOptions = toneToOptions(tone); // { tone: 'professional'|'relaxed'|'fun' }
  // You can also tune brevity based on UI later:
  const result = await aiInvoke<EmailGenerationInput, EmailGenerationOutput>('outreach.email', { ...input }, { tone: toneOptions.tone, brevity: opts?.brevity ?? 'medium' });
  return result;
}
