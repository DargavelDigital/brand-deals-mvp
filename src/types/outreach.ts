export type OutreachTone = 'professional' | 'relaxed' | 'fun'

export interface OutreachSequenceSettings {
  pauseFirstSend?: boolean
  replyDetection?: boolean
  autoFollowUp?: boolean
  tone?: OutreachTone
}

export interface OutreachSequence {
  name: string
  steps: any[]
  settings: OutreachSequenceSettings
}
