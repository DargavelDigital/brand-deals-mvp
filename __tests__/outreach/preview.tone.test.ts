import { describe, it, expect, vi } from 'vitest'
import { generateOutreachEmail } from '../../src/services/outreach/generateEmail'

vi.mock('../../src/ai/invoke', () => ({
  aiInvoke: vi.fn(async (_k, _input, opts) => {
    if (opts?.tone === 'professional') {
      return { 
        subject: 'Partnership opportunity', 
        body: 'Hi ACME team, I hope this email finds you well. I am reaching out regarding a potential collaboration opportunity that aligns with your brand values and our shared audience.', 
        toneUsed: 'professional' 
      }
    }
    if (opts?.tone === 'relaxed') {
      return { 
        subject: 'Quick idea for ACME', 
        body: 'Hey ACME team â€” hope you\'re doing great! I wanted to reach out about a fun collaboration idea that I think could really work for both of us.', 
        toneUsed: 'relaxed' 
      }
    }
    return { 
      subject: 'Fun collab idea ðŸŽ‰', 
      body: 'Hey ACME team ðŸ˜„ hope you\'re having an awesome day! I\'ve got this super exciting collaboration idea that I think could be absolutely amazing for both of us!', 
      toneUsed: 'fun' 
    }
  }),
}))

describe('Tone output differences', () => {
  const base = { 
    creator: { name: 'A', niche: 'tech', country: 'US', followers: 1000 }, 
    brand: { name: 'ACME', industry: 'tech', category: 'software', country: 'US' }, 
    sequence: { settings: {} } 
  }

  it('professional vs relaxed vs fun differ', async () => {
    const pro = await generateOutreachEmail({}, { ...base, sequence: { settings: { tone: 'professional' } } })
    const rel = await generateOutreachEmail({}, { ...base, sequence: { settings: { tone: 'relaxed' } } })
    const fun = await generateOutreachEmail({}, { ...base, sequence: { settings: { tone: 'fun' } } })

    expect(pro.body).not.toEqual(rel.body)
    expect(rel.body).not.toEqual(fun.body)
    expect(pro.body).not.toEqual(fun.body)

    expect(pro).toMatchSnapshot('professional')
    expect(rel).toMatchSnapshot('relaxed')
    expect(fun).toMatchSnapshot('fun')
  })

  it('defaults to professional tone when none specified', async () => {
    const result = await generateOutreachEmail({}, base)
    expect(result.toneUsed).toBe('professional')
  })

  it('uses sequence settings tone when provided', async () => {
    const result = await generateOutreachEmail({}, { ...base, sequence: { settings: { tone: 'fun' } } })
    expect(result.toneUsed).toBe('fun')
  })
})
