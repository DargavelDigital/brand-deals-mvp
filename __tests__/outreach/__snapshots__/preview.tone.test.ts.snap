// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Tone output differences > professional vs relaxed vs fun differ > fun 1`] = `
{
  "body": "Hey ACME team ðŸ˜„ hope you're having an awesome day! I've got this super exciting collaboration idea that I think could be absolutely amazing for both of us!",
  "subject": "Fun collab idea ðŸŽ‰",
  "toneUsed": "fun",
}
`;

exports[`Tone output differences > professional vs relaxed vs fun differ > professional 1`] = `
{
  "body": "Hi ACME team, I hope this email finds you well. I am reaching out regarding a potential collaboration opportunity that aligns with your brand values and our shared audience.",
  "subject": "Partnership opportunity",
  "toneUsed": "professional",
}
`;

exports[`Tone output differences > professional vs relaxed vs fun differ > relaxed 1`] = `
{
  "body": "Hey ACME team â€” hope you're doing great! I wanted to reach out about a fun collaboration idea that I think could really work for both of us.",
  "subject": "Quick idea for ACME",
  "toneUsed": "relaxed",
}
`;
